docs/nextra/data/blocks.rst

```diff
@@ -39,3 +39,27 @@ If you want the raw data bytes of a :py:class:`Block<steamship.data.block.Block>
 This is useful if you wish to share a generated image or audio file, or must make the content viewable in a place that cannot
 retain your Steamship API key.  You can also change the value of the ``public_data`` flag on an existing :py:class:`Block<steamship.data.block.Block>` by calling
 ``Block.set_public_data``.
+
+Streaming Blocks
+----------------
+
+A :py:class:`Block<steamship.data.block.Block>`'s content is immutable once created, but during the creation of a block its contents can be streamed.
+This stream can be consumed via a
+:class:`ServerSentEventStream<steamship.data.stream.ServerSentEventStream>`.
+
+* At the HTTP Level, this is implemented as Server-Sent Events (SSE)
+* At the Python SDK and Typescript SDK levels, this is implemented in callbacks about block updates.
+
+For a block, that stream will consist of :py:class:`BlockEvent<steamship.data.block.BlockEvent>` objects.
+These have the following possible :py:class:`BlockEventType<steamship.data.block.BlockEventType>`:
+
+* ``CREATED`` - The :py:class:`Block<steamship.data.block.Block>` has been created on a :py:class:`File<steamship.data.file.File>`
+* ``DATA_APPENDED`` - New data has been appended to the :py:class:`Block<steamship.data.block.Block>`
+* ``FINISHED`` - The :py:class:`Block<steamship.data.block.Block>` is fully formed and now immutable.
+
+For ``CREATED`` and ``FINISHED`` events, the payload will be the :py:class:`Block<steamship.data.block.Block>` object
+itself. For ``DATA_APPENDED`` events, the data will be the bytes of the data that has just been appended.
+
+Once the ``FINISHED`` event event has been fired, the block is considered immutable.
+Subsequent requests to stream the block will succeed, but the only event sent on the stream will be the ``FINISHED``
+event with the fully copy of the completed block.
```

Comment:
Currently at least, there's no SSE stream for a Block, it's simply a stream of the bytes of the content.  This is what I meant by it being the same as consuming `raw` but with potential waiting on the stream.

I think this is preferable, because it does not require any special handling for Blocks to consume the byte stream.